cvat/apps/redis_handler/tests/test_redis_migrations.py

```diff
@@ -0,0 +1,112 @@
+from cvat.apps.redis_handler.utils import get_class_from_module
+
+from .utils import path_to_module
+
+WORKDIR = Path("cvat/apps")
+
+MIGRATION_DIR = MigrationLoader.REDIS_MIGRATIONS_DIR_NAME
+MIGRATION_CLASS_NAME = MigrationLoader.REDIS_MIGRATION_CLASS_NAME
+MIGRATION_FILES = f"./**/{MIGRATION_DIR}/[0-9]*.py"
+
+MIGRATION_NAME_FORMAT = "{:03}_{}.py"
+BAD_MIGRATION_FILE = """\
+class Migration:
+    @classmethod
+    def run(cls): ...
+
+"""
+
+
+@patch(
+    f"cvat.apps.redis_handler.management.commands.migrateredis.Redis",
+    return_value=fakeredis.FakeRedis(),
+)
+class TestRedisMigrations(TestCase):
+    class BadMigration:
+
+        def __init__(self, app_name: str, migration_name: str):
+            self.app_name = app_name
+            self.app_path = WORKDIR / app_name
+            assert self.app_path.exists()
+            self.migration_name = migration_name
+            self.number = 0
+
+            self.migration_file_path = self.generate_migration_file()
+            mock_migration_module_path = path_to_module(self.migration_file_path)
+
+            self.test_class = get_class_from_module(
+                mock_migration_module_path, MIGRATION_CLASS_NAME
+            )
+            assert self.test_class is not None
+            self.patcher = patch.object(self.test_class, "run")
+            self.runner_mock: Mock | None = self.patcher.start()
+
+        def make_migration_name(self):
+            return MIGRATION_NAME_FORMAT.format(self.number, self.migration_name)
+
+        def generate_migration_file(self) -> Path:
+            migration_dir = self.app_path / MIGRATION_DIR
+            if not os.path.exists(migration_dir):
+                os.mkdir(migration_dir)
+            filename = self.make_migration_name()
+            migration_file_path = migration_dir / filename
+            with open(migration_file_path, "w") as file:
+                file.write(BAD_MIGRATION_FILE)
+            return migration_file_path
+
+        def cleanup(self):
+            self.patcher.stop()
+            self.runner_mock = None
+            os.remove(self.migration_file_path)
+
+    def test_migration_added_and_applied(self, redis):
+
+        def file_to_migration_name(path: Path) -> str:
+            name = path_to_module(path)
+            name = name.removeprefix("cvat.apps.")
+            name = name.replace("redis_migrations.", "")
+            return name
+
+        # Keys are not added yet
+        with self.assertRaises(SystemExit):
+            call_command("migrateredis", check=True)
+
+        # Add keys
+        call_command("migrateredis")
+
+        # Keys are added
+        self.assertIsNone(call_command("migrateredis", check=True))
+
+        # Check keys added
+        migration_files = WORKDIR.glob(MIGRATION_FILES)
+        expected_migrations = set(
+            file_to_migration_name(file).encode("utf8") for file in migration_files
+        )
```

Comment:
It's better to use existing logic instead of adding custom `file_to_migration_name`
```suggestion
        expected_migrations = {
            f"{app_config.label}.{f.stem}".encode()
            for app_config in MigrationLoader.find_app_configs()
            for f in (Path(app_config.path) / MigrationLoader.REDIS_MIGRATIONS_DIR_NAME).glob("[0-9]*.py")
        }
        assert len(expected_migrations)
```